test/e2e/network/dns_common.go

```diff
@@ -235,6 +248,21 @@ func (t *dnsTestCommon) deleteUtilPod() {
 	}
 }
 
+// deleteCoreDNSPods manually deletes the CoreDNS pods to apply the changes to the ConfigMap.
+func (t *dnsTestCommon) deleteCoreDNSPods() {
+
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"k8s-app": "kube-dns"}))
+	options := metav1.ListOptions{LabelSelector: label.String()}
+
+	pods, err := t.f.ClientSet.CoreV1().Pods("kube-system").List(options)
+	podClient := t.c.CoreV1().Pods(metav1.NamespaceSystem)
+
+	for _, pod := range pods.Items {
+		err = podClient.Delete(pod.Name, metav1.NewDeleteOptions(0))
+		Expect(err).NotTo(HaveOccurred())
+	}
+}
```

Comment:
Should you wait the Pod is restored before going ahead ? or that is part of a delete option ?